### PR TITLE
store logs in seperate directory

### LINKED ISSUE #136: Make all launcher logs go into a 'logs' directory
### Role

I am a Prism Launcher user.

### Suggestion

I want there to be a dedicated logs folder, instead of launcher logs being placed into the root of the settings directory. For example, the Prism Launcher AppData folder on Windows.

### Benefit

So that it is more organised and clean.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

image

### PR DESCRIPTION
**Pull request by Trial97**
Revive https://github.com/PrismLauncher/PrismLauncher/pull/678
Same PR with the mentioned changes was done.

closes https://github.com/PrismLauncher/PrismLauncher/issues/136.

### CODE REVIEW

**Comment by flowln** [comment]
I think you and @RedsonBr140 should decide on which PR to use, since this duplicates #1064 (though you did cherry-pick the original commit, so I favor this PR if there's a tie). Also, please note the convo in there, I still think a migration is easy to do and would be good.

**Comment by RedsonBr140** [comment]
> I think you and @RedsonBr140 should decide on which PR to use, since this duplicates #1064 (though you did cherry-pick the original commit, so I favor this PR if there's a tie). Also, please note the convo in there, I still think a migration is easy to do and would be good.

Sure. I'll do ASAP. I'm experiencing some trouble with my computer right now, I may solve them in one week and work this PR out.

**Comment by RedsonBr140** [comment]
> I think you and @RedsonBr140 should decide on which PR to use, since this duplicates #1064 (though you did cherry-pick the original commit, so I favor this PR if there's a tie). Also, please note the convo in there, I still think a migration is easy to do and would be good.

Sure. I'll do ASAP. I'm experiencing some trouble with my computer right now, I may solve them in one week and work this PR out.

**Comment by Trial97** [comment]
Well, I also added a migration for the files :D

File: launcher/Application.cpp [hunk 1]
```diff
@@ -376,9 +376,18 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
...
-            "logs");  // this can fail, but there is no need to throw an error *yet*, since it also triggers the error message below!
+        static const QString baseLogFile = BuildConfig.LAUNCHER_NAME + "-%0.log";
+        static const QString logBase = FS::PathCombine("logs", baseLogFile);
+        if (FS::ensureFolderPathExists("logs")) {  // if this did not fail
+            for (auto i = 0; i <= 4; i++) {
+                auto oldName = baseLogFile.arg(i);
+                auto newName = logBase.arg(i);
+                if (QFile::exists(newName))  // in case there are already files in folder just to be safe add a suffix
+                    newName += ".old";
+                QFile::rename(oldName, newName);
```
Comment (flowln) [inline launcher/Application.cpp]: move all of this down below and use the `moveFile` function. Also, no need to add the suffix. As I said, no users will have anything in the `logs` folder, they won't even have the folder.

I think it also makes sense to either use the format already used

```cpp
moveFile(logBase.arg(3), logBase.arg(4));
moveFile(logBase.arg(2), logBase.arg(3));
moveFile(logBase.arg(1), logBase.arg(2));
moveFile(logBase.arg(0), logBase.arg(1));
```

, or convert that to a for loop.

**Comment by RedsonBr140** [comment]
> Well, I also added a migration for the files :D

LoL didn't see this was a brand new PR. I think it may be better to merge this one as it's already done @flowln. If you do so, I can close my PR with a comment about what happened.

File: launcher/Application.cpp (hunk 1 above)
Comment (Trial97) [reply launcher/Application.cpp]: I fixed this, but still, I kept an existing verification for the old log file in order not to always remove the new files. (without it on each launch the previous logs were deleted even if no old logs were present)

**Review by Ryex — APPROVED** [review]
LGTM

No suggestions.

**Review by getchoo — APPROVED** [review]
ditto

**Review by DioEgizio — APPROVED** [review]

**Comment by getchoo** [comment]
thanks!

**Pull request merged**